### ISSUE TITLE
Precompiling Oscar ERROR: LoadError: UndefVarError: libsingular not defined

### ISSUE THREAD

**Issue by yusufferdogan** [oscar-system/Oscar.jl#990]
```
julia> using Oscar
[ Info: Precompiling Oscar [f1435218-dba5-11e9-1e4d-f1a5fab5fc13]
ERROR: LoadError: UndefVarError: libsingular not defined
Stacktrace:
 [1] getproperty(x::Module, f::Symbol)
   @ Base .\Base.jl:26
 [2] top-level scope
   @ C:\Users\yusuf\.julia\packages\Singular\gc8mN\src\Singular.jl:66
 [3] include
   @ .\Base.jl:386 [inlined]
 [4] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::String)
   @ Base .\loading.jl:1235
 [5] top-level scope
   @ none:1
 [6] eval
   @ .\boot.jl:360 [inlined]
 [7] eval(x::Expr)
   @ Base.MainInclude .\client.jl:446
 [8] top-level scope
   @ none:1
in expression starting at C:\Users\yusuf\.julia\packages\Singular\gc8mN\src\Singular.jl:1
```

**Comment by thofma** [comment]
Oscar is not working on Windows natively. There are instructions over at https://oscar.computeralgebra.de/install/ on how to use it on Windows.

**Comment by tthsqe12** [comment]
But at least we can give a better error message?

**Comment by thofma** [comment]
Do you have an idea? It used to be better I believe, since we added:
https://github.com/oscar-system/Oscar.jl/blob/8c787842e75df4039e0f3e7daf0f9a634a49644a/src/Oscar.jl#L37-L39
But apparently it loads first `Singular.jl`, which tries to open the shared library. I am not sure what we can do there.

**Comment by yusufferdogan** [comment]
```
using Oscar
[ Info: Precompiling Oscar [f1435218-dba5-11e9-1e4d-f1a5fab5fc13]

signal (6): Aborted
in expression starting at /home/yusuf/.julia/packages/Oscar/Bh1Um/src/Oscar.jl:33
gsignal at /usr/bin/../lib/x86_64-linux-gnu/libc.so.6 (unknown line)
abort at /usr/bin/../lib/x86_64-linux-gnu/libc.so.6 (unknown line)
unknown function (ip: 0x7f6415a07b61)
jl_method_table_insert at /usr/bin/../lib/x86_64-linux-gnu/libjulia.so.1 (unknown line)
unknown function (ip: 0x7f6415aaa447)
jl_restore_incremental at /usr/bin/../lib/x86_64-linux-gnu/libjulia.so.1 (unknown line)
_include_from_serialized at ./loading.jl:681
_require_search_from_serialized at ./loading.jl:781
_require at ./loading.jl:1006
require at ./loading.jl:927
require at ./loading.jl:922
unknown function (ip: 0x7f6415ab3113)
unknown function (ip: 0x7f6415ab4382)
unknown function (ip: 0x7f6415ab478e)
unknown function (ip: 0x7f6415ab413a)
unknown function (ip: 0x7f6415ab43c8)
unknown function (ip: 0x7f6415a8f105)
jl_load at /usr/bin/../lib/x86_64-linux-gnu/libjulia.so.1 (unknown line)
include at ./Base.jl:377
unknown function (ip: 0x7f6415a9975b)
unknown function (ip: 0x7f6415a99389)
unknown function (ip: 0x7f6415a998f0)
unknown function (ip: 0x7f6415a9a9c8)
unknown function (ip: 0x7f6415a9afd9)
unknown function (ip: 0x7f6415a9b616)
unknown function (ip: 0x7f6415ab3e08)
jl_toplevel_eval_in at /usr/bin/../lib/x86_64-linux-gnu/libjulia.so.1 (unknown line)
eval at ./boot.jl:331 [inlined]
eval at ./client.jl:449
top-level scope at ./none:3
unknown function (ip: 0x7f6415ab4030)
unknown function (ip: 0x7f6415ab43c8)
jl_toplevel_eval_in at /usr/bin/../lib/x86_64-linux-gnu/libjulia.so.1 (unknown line)
eval at ./boot.jl:331
exec_options at ./client.jl:264
_start at ./client.jl:484
jfptr__start_2075.clone_1 at /usr/lib/x86_64-linux-gnu/julia/sys.so (unknown line)
unknown function (ip: 0x7f6415f4a9d9)
unknown function (ip: 0x7f6415f4a5a6)
__libc_start_main at /usr/bin/../lib/x86_64-linux-gnu/libc.so.6 (unknown line)
unknown function (ip: 0x7f6415f4a64d)
Allocations: 25882691 (Pool: 25875179; Big: 7512); GC: 23
ERROR: Failed to precompile Oscar [f1435218-dba5-11e9-1e4d-f1a5fab5fc13] to /home/yusuf/.julia/compiled/v1.4/Oscar/aJ3Pg_p23DB.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] compilecache(::Base.PkgId, ::String) at ./loading.jl:1272
 [3] _require(::Base.PkgId) at ./loading.jl:1029
 [4] require(::Base.PkgId) at ./loading.jl:927
 [5] require(::Module, ::Symbol) at ./loading.jl:922
```

**Comment by yusufferdogan** [comment]
When I tried this on Ubuntu, it gave me this
I followed the exact instructions.

**Comment by yusufferdogan** [comment]
OUTPUT  OF Pkg.add("Oscar")
```
Pkg.add("Oscar")
   Updating registry at `~/.julia/registries/General`
   Updating git-repo `https://github.com/JuliaRegistries/General.git`
  Resolving package versions...
  Installed GMP_jll ────────────────────── v6.1.2+6
  Installed Zstd_jll ───────────────────── v1.4.8+0
  Installed Ncurses_jll ────────────────── v6.1.0+5
  Installed Libiconv_jll ───────────────── v1.16.0+8
  Installed FLINT_jll ──────────────────── v200.800.400+0
  Installed CxxWrap ────────────────────── v0.11.2
  Installed AbstractAlgebra ────────────── v0.21.0
  Installed MPFR_jll ───────────────────── v4.0.2+4
  Installed Arb_jll ────────────────────── v200.2000.0+0
  Installed Singular_jll ───────────────── v402.100.101+0
  Installed cddlib_jll ─────────────────── v0.94.10+1
  Installed BinaryProvider ─────────────── v0.5.10
  Installed JSON ───────────────────────── v0.21.2
  Installed Oscar ──────────────────────── v0.6.0
  Installed lrslib_jll ─────────────────── v0.3.2+0
  Installed MozillaCACerts_jll ─────────── v2021.1.19+0
  Installed Readline_jll ───────────────── v8.0.4+1
  Installed LZO_jll ────────────────────── v2.10.0+3
  Installed CompilerSupportLibraries_jll ─ v0.3.4+0
  Installed GAP_lib_jll ────────────────── v400.1190.200+0
  Installed Mongoc ─────────────────────── v0.6.2
  Installed Nemo ───────────────────────── v0.26.1
  Installed Perl_jll ───────────────────── v5.30.3+2
  Installed libsingular_julia_jll ──────── v0.17.4+0
  Installed XML2_jll ───────────────────── v2.9.10+3
  Installed PPL_jll ────────────────────── v1.2.0+1
  Installed Ninja_jll ──────────────────── v1.10.2+0
  Installed LoadFlint ──────────────────── v0.6.4
  Installed Polymake ───────────────────── v0.5.8
  Installed polymake_jll ───────────────── v400.400.2+0
  Installed Gettext_jll ────────────────── v0.20.1+7
  Installed snappy_jll ─────────────────── v1.1.8+1
  Installed GAP_jll ────────────────────── v400.1190.205+0
  Installed libpolymake_julia_jll ──────── v0.4.304+0
  Installed JLLWrappers ────────────────── v1.4.0
  Installed nauty_jll ──────────────────── v2.6.12+1
  Installed Artifacts ──────────────────── v1.3.0
  Installed libcxxwrap_julia_jll ───────── v0.8.4+0
  Installed bliss_jll ──────────────────── v0.73.0+1
  Installed normaliz_jll ───────────────── v300.900.0+0
  Installed boost_jll ──────────────────── v1.71.0+3
  Installed TOML ───────────────────────── v1.0.3
  Installed GAP ────────────────────────── v0.6.2
  Installed Hecke ──────────────────────── v0.10.22
  Installed Antic_jll ──────────────────── v0.200.500+0
  Installed lib4ti2_jll ────────────────── v1.6.9+2
  Installed GLPK_jll ───────────────────── v5.0.0+0
  Installed MongoC_jll ─────────────────── v1.16.2+1
  Installed Parsers ────────────────────── v2.2.0
  Installed Preferences ────────────────── v1.2.3
  Installed OpenSSL_jll ────────────────── v1.1.1+6
  Installed Calcium_jll ────────────────── v0.400.100+0
  Installed Requires ───────────────────── v1.3.0
  Installed MacroTools ─────────────────── v0.5.9
  Installed DocStringExtensions ────────── v0.8.6
  Installed Zlib_jll ───────────────────── v1.2.11+18
  Installed Singular ───────────────────── v0.6.0
Downloading artifact: Zstd
######################################################################## 100.0%#=#=-#  #                Downloading artifact: Ncurses
######################################################################## 100.0%-=#=#   #   #            Downloading artifact: Libiconv
######################################################################## 100.0%#=#=-#  #                Downloading artifact: GMP
######################################################################## 100.0%#=#=-#  #                Downloading artifact: FLINT
######################################################################## 100.0%#=#=-#  #                Downloading artifact: Arb
######################################################################## 100.0%#=#=-#  #                Downloading artifact: Singular
######################################################################## 100.0%-#O#- #   #              Downloading artifact: cddlib
######################################################################## 100.0%#=#=-#  #                Downloading artifact: MPFR
######################################################################## 100.0%#=#=-#  #                Downloading artifact: lrslib
######################################################################## 100.0%-#O#- #   #              Downloading artifact: GAP_lib
######################################################################## 100.0%#=#=-#  #                Downloading artifact: CompilerSupportLibraries
######################################################################## 100.0%#=#=-#  #                Downloading artifact: LZO
######################################################################## 100.0%#=#=-#  #                Downloading artifact: Readline
######################################################################## 100.0%#=#=-#  #                Downloading artifact: MozillaCACerts
Downloading artifact: libsingular_julia
######################################################################## 100.0%#=#=-#  #                Downloading artifact: Perl
######################################################################## 100.0%-=O=-                    Downloading artifact: XML2
######################################################################## 100.0%#=#=-#  #                Downloading artifact: PPL
######################################################################## 100.0%#=#=-#  #                Downloading artifact: Ninja
######################################################################## 100.0%#=#=-#  #                Downloading artifact: polymake
######################################################################## 100.0%-#O#- #   #              Downloading artifact: snappy
######################################################################## 100.0%-#O#- #   #              Downloading artifact: GAP
######################################################################## 100.0%#=#=-#  #                Downloading artifact: Gettext
######################################################################## 100.0%#=#=-#  #                Downloading artifact: SmallGroupDB
######################################################################## 100.0%#=#=-#  #                Downloading artifact: libpolymake_julia
######################################################################## 100.0%#=#=-#  #                Downloading artifact: nauty
######################################################################## 100.0%-=O=-                    Downloading artifact: libcxxwrap_julia
######################################################################## 100.0%#=#=-#  #                Downloading artifact: bliss
######################################################################## 100.0%#=#=-#  #                Downloading artifact: normaliz
######################################################################## 100.0%#=#=-#  #                Downloading artifact: boost
######################################################################## 100.0%-=#=#   #   #            Downloading artifact: gap_packages
######################################################################## 100.0%-=O=-                    Downloading artifact: Antic
######################################################################## 100.0%#=#=-#  #                Downloading artifact: Zlib
######################################################################## 100.0%#=#=-#  #                Downloading artifact: lib4ti2
######################################################################## 100.0%#=#=-#  #                Downloading artifact: GLPK
######################################################################## 100.0%#=#=-#  #                Downloading artifact: MongoC
Downloading artifact: OpenSSL
######################################################################## 100.0%#=#=-#  #                Downloading artifact: Calcium
######################################################################## 100.0%#=#=-#  #                   Updating `~/.julia/environments/v1.4/Project.toml`
  [c3fe647b] ↓ AbstractAlgebra v0.22.2 ⇒ v0.21.0
  [f1435218] + Oscar v0.6.0
   Updating `~/.julia/environments/v1.4/Manifest.toml`
  [c3fe647b] ↓ AbstractAlgebra v0.22.2 ⇒ v0.21.0
  [e21ec000] + Antic_jll v0.200.500+0
  [d9960996] + Arb_jll v200.2000.0+0
  [56f22d72] + Artifacts v1.3.0
  [b99e7846] + BinaryProvider v0.5.10
  [fcfa6d1b] + Calcium_jll v0.400.100+0
  [e66e0078] + CompilerSupportLibraries_jll v0.3.4+0
  [1f15a43c] + CxxWrap v0.11.2
  [ffbed154] + DocStringExtensions v0.8.6
  [e134572f] + FLINT_jll v200.800.400+0
  [c863536a] + GAP v0.6.2
  [5cd7a574] + GAP_jll v400.1190.205+0
  [de1ad85e] + GAP_lib_jll v400.1190.200+0
  [e8aa6df9] + GLPK_jll v5.0.0+0
  [781609d7] + GMP_jll v6.1.2+6
  [78b55507] + Gettext_jll v0.20.1+7
  [d5909c97] - GroupsCore v0.4.0
  [3e1990a7] + Hecke v0.10.22
  [692b3bcd] + JLLWrappers v1.4.0
  [682c06a0] + JSON v0.21.2
  [dd4b983a] + LZO_jll v2.10.0+3
  [94ce4f54] + Libiconv_jll v1.16.0+8
  [472f376f] + LoadFlint v0.6.4
  [3a97d323] + MPFR_jll v4.0.2+4
  [1914dd2f] + MacroTools v0.5.9
  [90100e71] + MongoC_jll v1.16.2+1
  [4fe8b98c] + Mongoc v0.6.2
  [14a3606d] + MozillaCACerts_jll v2021.1.19+0
  [68e3532b] + Ncurses_jll v6.1.0+5
  [2edaba10] + Nemo v0.26.1
  [76642167] + Ninja_jll v1.10.2+0
  [458c3c95] + OpenSSL_jll v1.1.1+6
  [f1435218] + Oscar v0.6.0
  [80dd9cbb] + PPL_jll v1.2.0+1
  [69de0a69] + Parsers v2.2.0
  [83958c19] + Perl_jll v5.30.3+2
  [d720cf60] + Polymake v0.5.8
  [21216c6a] + Preferences v1.2.3
  [05236dd9] + Readline_jll v8.0.4+1
  [ae029012] + Requires v1.3.0
  [bcd08a7b] + Singular v0.6.0
  [43d676ae] + Singular_jll v402.100.101+0
  [fa267f1f] + TOML v1.0.3
  [02c8fc9c] + XML2_jll v2.9.10+3
  [83775a58] + Zlib_jll v1.2.11+18
  [3161d3a3] + Zstd_jll v1.4.8+0
  [508c9074] + bliss_jll v0.73.0+1
  [28df3c45] + boost_jll v1.71.0+3
  [f07e07eb] + cddlib_jll v0.94.10+1
  [1493ae25] + lib4ti2_jll v1.6.9+2
  [3eaa8342] + libcxxwrap_julia_jll v0.8.4+0
  [4d8266f6] + libpolymake_julia_jll v0.4.304+0
  [ae4fbd8f] + libsingular_julia_jll v0.17.4+0
  [3873f7d0] + lrslib_jll v0.3.2+0
  [55c6dc9b] + nauty_jll v2.6.12+1
  [6690c6e9] + normaliz_jll v300.900.0+0
  [7c209550] + polymake_jll v400.400.2+0
  [fe1e1685] + snappy_jll v1.1.8+1
  [ade2ca70] + Dates
  [76f85450] + LibGit2
  [a63ad114] + Mmap
  [44cfe95a] + Pkg
  [de0858da] + Printf
  [9abbd945] + Profile
  [3fa0cd96] + REPL
  [ea8e919c] + SHA
  [10745b16] + Statistics
  [cf7118a7] + UUIDs
  [4ec0a83e] + Unicode
   Building LoadFlint → `~/.julia/packages/LoadFlint/ZaVVM/deps/build.log`
   Building Nemo ─────→ `~/.julia/packages/Nemo/EXnlj/deps/build.log
```

**Comment by thofma** [comment]
How did you install julia within ubuntu?

**Comment by fingolfin** [comment]
This is Oscar 0.6.0 and Julia 1.4 which both are outdated and not supported anymore

**Comment by fingolfin** [comment]
So most likely they installed Julia via the package manager. Not good

**Comment by fingolfin** [comment]
The website actually was still saying that the minimum Julia version is 1.3.1; I've fixed this now

**Comment by yusufferdogan** [comment]
> How did you install julia within ubuntu?

yes sir

**Comment by yusufferdogan** [comment]
> How did you install julia within ubuntu?

https://sebasguts.github.io/oscar-website/install/ 

like this sir

**Comment by yusufferdogan** [comment]
```
julia> Pkg.add(PackageSpec(url="https://github.com/oscar-system/Singular.jl", rev="master" ))
  Updating git-repo `https://github.com/oscar-system/Singular.jl`
 Resolving package versions...
┌ Warning: julia version requirement for package Singular not satisfied
└ @ Pkg.Operations /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.1/Pkg/src/Operations.jl:154
ERROR: Unsatisfiable requirements detected for package CxxWrap [1f15a43c]:
 CxxWrap [1f15a43c] log:
 ├─possible versions are: [0.7.0-0.7.4, 0.8.0-0.8.2, 0.9.0-0.9.1, 0.10.0-0.10.2, 0.11.0-0.11.2, 0.12.0] or uninstalled
 ├─restricted to versions [0.11.0-0.11, 0.12.0-0.12] by Singular [bcd08a7b], leaving only versions [0.11.0-0.11.2, 0.12.0]
 │ └─Singular [bcd08a7b] log:
 │   ├─possible versions are: 0.9.1 or uninstalled
 │   └─Singular [bcd08a7b] is fixed to version 0.9.1
 └─restricted to versions 0.9.1 by an explicit requirement — no versions left
```

**Comment by thofma** [comment]
As I said above, the official installation guide is at https://oscar.computeralgebra.de/install/.
